storage/remote/store.go

```diff
@@ -0,0 +1,141 @@
+package remote
+
+import (
+	"context"
+	"fmt"
+	"github.com/onflow/flow-archive/api/archive"
+	"github.com/onflow/flow-archive/codec/zbor"
+	"github.com/onflow/flow-emulator/storage"
+	"github.com/onflow/flow-emulator/storage/sqlite"
+	exeState "github.com/onflow/flow-go/engine/execution/state"
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/fvm/storage/snapshot"
+	"github.com/onflow/flow-go/ledger/common/pathfinder"
+	"github.com/onflow/flow-go/ledger/complete"
+	flowgo "github.com/onflow/flow-go/model/flow"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type Store struct {
+	*sqlite.Store
+	client archive.APIClient
+}
+
+func New(chainID flowgo.ChainID) (*Store, error) {
+	archiveHosts := map[flowgo.ChainID]string{
+		flowgo.Mainnet: "archive.mainnet.nodes.onflow.org:9000",
+		flowgo.Testnet: "archive.testnet.nodes.onflow.org:9000",
+	}
+
+	host, ok := archiveHosts[chainID]
+	if !ok {
+		return nil, fmt.Errorf("chain %s not supported with remote store", chainID.String())
+	}
+
+	conn, err := grpc.Dial(
+		host,
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not connect to archive node")
+	}
+
+	memorySql, err := sqlite.New(sqlite.InMemory)
+	if err != nil {
+		return nil, err
+	}
+
+	store := &Store{
+		client: archive.NewAPIClient(conn),
+		Store:  memorySql,
+	}
+
+	store.DataGetter = store
+	store.DataSetter = store
+	store.KeyGenerator = &storage.DefaultKeyGenerator{}
+
+	return store, nil
+}
+
+func (s *Store) BlockByID(ctx context.Context, blockID flowgo.Identifier) (*flowgo.Block, error) {
+	// todo use local storage first as a cache
+
+	heightRes, err := s.client.GetHeightForBlock(ctx, &archive.GetHeightForBlockRequest{BlockID: blockID[:]})
+	if err != nil {
+		return nil, err
+	}
+
+	return s.BlockByHeight(ctx, heightRes.Height)
+}
+
+func (s *Store) LatestBlock(ctx context.Context) (flowgo.Block, error) {
+	heightRes, err := s.client.GetLast(ctx, &archive.GetLastRequest{})
+	if err != nil {
+		return flowgo.Block{}, err
+	}
+
+	block, err := s.BlockByHeight(ctx, heightRes.Height)
+	if err != nil {
+		return flowgo.Block{}, err
+	}
+
+	return *block, nil
+}
+
+func (s *Store) BlockByHeight(ctx context.Context, height uint64) (*flowgo.Block, error) {
+	// todo use local storage first as a cache
```

Comment:
Will this be a separate PR?

Without this you cannot make changes with local transactions, right?

Reply:
You can because this function is not used for that, the get ledger one is which first checks local. I would make it separate PR just to keep it small.